### ISSUE TITLE
Invalid quoting of UTF-8 character 

### ISSUE THREAD

**Issue by m2ym** [janestreet/sexplib#11]
If an atom contains UTF-8 character and a whitespace, then the output will be corrupted.

Here is example.

```OCaml
open Core_kernel.Std

let () =
  Format.printf "%a@." Sexp.pp_mach (Sexp.of_string "(String\"\\233\\171\\152\")");
  Format.printf "%a@." Sexp.pp_mach (Sexp.of_string "(String\"\\233\\171\\152 Hello\")")
```

UTF-8 character `#E9AB98` is `高`, so the expected output is

```
(String 高)
(String "高 Hello")
```

but the actual output is

```
(String 高)
(String"152 Hello")
```

The second line is obviously wrong.

**Comment by ejgallego** [comment]
This seems to be fixed, (see #18 ) ?

**Comment by diml** [comment]
Indeed this is now fixed
